Classes/Private/NSAttributedString/Converters/HTMLNodeToNSAttributedString.swift

```diff
@@ -161,62 +155,55 @@ class HMTLNodeToNSAttributedString: SafeConverter {
+    /// Calculates the attributes for this node based.  Returns a dictionary which includes
+    /// inherited attributes.
     ///
     /// - Parameters:
-    ///     - string: the string to add the attributes to.  This string will be modified in-place
-    ///             to add the necessary attributes.
     ///     - node: the node to get the information from.
     ///
-    private func addAttributes(toString string: NSMutableAttributedString, fromNode node: ElementNode) {
+    private func attributes(forNode node: ElementNode, inheritingAttributes inheritedAttributes: [String:AnyObject]) -> [String:AnyObject] {
 
-        let fullRange = NSRange(location: 0, length: string.length)
+        var attributes = inheritedAttributes
 
-        addStyleAttributes(toString: string, fromNode: node)
-        string.addAttribute(keyForNode(node), value: node, range: fullRange)
-    }
+        // Since a default font is requested by this class, there's no way this attribute should
+        // ever be unset.
+        //
+        precondition(inheritedAttributes[NSFontAttributeName] is UIFont)
+        let baseFont = inheritedAttributes[NSFontAttributeName] as! UIFont
+        let baseFontDescriptor = baseFont.fontDescriptor()
+        let descriptor = fontDescriptor(forNode: node, withBaseFontDescriptor: baseFontDescriptor)
 
-    /// Adds all style attributes from the specified element node to the specified string.
-    /// You'll usually want to call `addAttributes(toString:fromNode:)` instead of this method.
-    ///
-    /// - Parameters:
-    ///     - string: the string to apply the styles to.  This string will be modified in-place
-    ///             to add the necessary attributes.
-    ///     - node: the node to get the style information from.
-    ///
-    private func addStyleAttributes(toString string: NSMutableAttributedString, fromNode node: ElementNode) {
+        if descriptor != baseFontDescriptor {
+            attributes[NSFontAttributeName] = UIFont(descriptor: descriptor, size: descriptor.pointSize)
+        }
 
-        let fullRange = NSRange(location: 0, length: string.length)
+        if isLink(node) {
+            let linkURL: String
 
-        if isUnderlined(node) {
-            string.addAttribute(NSUnderlineStyleAttributeName, value: 1, range: fullRange)
+            if let attribute = node.attributes.indexOf({ $0.name == "href" }) as? Libxml2.HTML.StringAttribute {
+                linkURL = attribute.value
+            } else {
+                // We got a link tag without an HREF attribute
+                //
+                linkURL = ""
+            }
+
+            attributes[NSLinkAttributeName] = linkURL
         }
 
         if isStrikedThrough(node) {
-            string.addAttribute(NSStrikethroughStyleAttributeName, value: 1, range: fullRange)
+            attributes[NSStrikethroughStyleAttributeName] = 1
+        }
+
+        if isUnderlined(node) {
+            attributes[NSUnderlineStyleAttributeName] = 1
```

Comment:
Maybe `NSUnderlineStyle.StyleSingle.rawValue` instead of `1`, for both `isStrikedThrough` and `isUnderlined` `if` blocks.